02/app.js

```diff
@@ -0,0 +1,52 @@
+const form = document.querySelector('form');
+const submit = document.querySelector('input[type = submit]');
+
+const registerConditions = function (e) {
+    e.preventDefault();
+    let email = form.elements[0].value;
+    let paswd1 = form.elements[1].value;
+    let paswd2 = form.elements[2].value;
+    const regulation = form.elements[3];
+    const checkboxLabel = document.querySelector('[for="formAccept"]');
+    const regex = /@/;
+    let correct = true;
+
+    if (regex.test(email)) {
+        form.elements[2].style.color = 'black';
+        if (paswd1 === paswd2) {
+            form.elements[2].style.color = 'black';
+            if (paswd1.length > 6) {
+                if (form.elements[3].checked) {
+                    checkboxLabel.style.color = 'black';
+                    console.log("done");
+                    for (const el of form.elements) {
+                        if (el.getAttribute('type') != 'submit') {
+                            el.value = '';
+                            form.elements[3].checked = false;
+                        }
+                    }
+                } else {
```

Comment:
Zagnieżdżanie ifów powoduje, że kod jest bardzo nieczytelny i łatwo się pomylić, gdzie jaki warunek się kończy.
Jak już to lepiej zrobić te elementy nie zależne i użyć dodatkowej zmiennej czyli:
```
if(regex.test(email)) {
} else {
    errors.push(...);
}

if(paswd1 === paswd2) {
} else {
     errors.push(...);
}

resetAllLabelsColor();
if(errors.length > 0) {
    showErrors(errors);
}
```